utils/s3/client.cc

```diff
@@ -173,9 +173,13 @@ future<> client::make_request(http::request req, http::experimental::client::rep
     auto it = _https.find(sg);
     if (it == _https.end()) [[unlikely]] {
         auto factory = std::make_unique<dns_connection_factory>(_host, _cfg->port, _cfg->use_https);
+        // Limit the maximum number of connections this group's http client
+        // may have proportional to its shares. Shares are typically in the
+        // range of 100...1000, thus resulting in 1..10 connections
+        auto max_connections = std::max((unsigned)(sg.get_shares() / 100), 1u);
         it = _https.emplace(std::piecewise_construct,
```

Comment:
you're "flooring" the result on purpose (in a conservative act), right?

Reply:
Yes. It's no way the final concurrency maintenance, in this PR I'm more interested in 
a) making concurrency per-sched-class
b) putting it all into s3 client

The exact number of sockets each class may have can change in the future. Even better -- seastar http client has `set_max_connections()` call that can change it on the fly